hardware/eHouse/EhouseEvents.cpp

```diff
@@ -0,0 +1,305 @@
+/* 
+ * File:   Events.cpp
+ * ONLY for usage with eHouse system controllers | Intellectual property of iSys.Pl
+ * eHouse4Ethernet Server - described in UdpListener.cpp
+ * eHouse1 (RS-485) Server  //described in rs485.cpp
+ * eHouse1 under CommManager Supervision Server - described in UdpListener.cpp
+ * TCP Client for submitting events to Ethernet eHouse Controllers and eHouse 1 via CommManager in case of eHouse under CM supervision version described in TcpClient.c
+ * 
+ * EventQueue Process 
+ * 
+ * Author: Robert Jarzabek, iSys.Pl
+ * http://en.isys.pl/          eHouse Home automation Producer Web Page
+ * http://eHouse.biz/      eHouse Home automation SHOP 
+ * http://smart.ehouse.pro/   eHouse Home Automation - Do it yourself, programming examples, designing
+ * Recently updated 11.11.2017
+ * globals.h - global configuration / main settings
+ * most important params for eHouse events manager
+ * 
+ * COMMANAGER_IP_HIGH - set CommManager Ip addr h if differs from 192.168.0.254 (192.168.%COMMANAGER_IP_HIGH%.%COMMANAGER_IP_LOW%)
+ * COMMANAGER_IP_LOW - set CommManager Ip addr l
+ * EHOUSE_PRO_HIGH - default eHouse Pro Server IP addr h (0)
+ * EHOUSE_PRO_LOW  - default eHouse Pro Server IP addr l (150)
+ * RE_SEND_RETRIES - No of retries in case of execution failure
+ * RE_TIME_REPLAY  - Timer for retry in (RE_TIME_RETRY-RE_SEND_RETRIES) *0.1s
+ * 
+ * Events.cpp - Event Queue processing 
+ */
+
+#include "stdafx.h"
+#include "../main/Logger.h"
+#include "../eHouseTCP.h"
+#ifndef WIN32
+
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+//#include <stropts.h>
+#include <unistd.h>
+#include <dirent.h>
+#include <net/if.h>
+//#include <linux/netdevice.h>
+#else
+#include <Windows.h>
+#endif
+#include "globals.h"
+#include "status.h"
+
+#include <string.h>
+#include <stdio.h>
+#include <fcntl.h>
+
+
+
+unsigned int EventsCountInQueue=0;      //Events In queue count to bypass processing EventQueue when it is empty
```

Comment:
Could you make this a member of the class ?
Same for
eHouseProN
ExecEvent
AddToLocalEvent